packages/ketcher-macromolecules/src/state/common/editorSlice.ts

```diff
@@ -98,16 +97,17 @@ export const editorSlice: Slice = createSlice({
         theme: DeepPartial<ThemeType>;
         canvas: SVGSVGElement;
         monomersLibraryUpdate?: string | JSON;
+        onInit?: (editor: CoreEditor) => void;
       }>,
     ) => {
-      state.editor = new CoreEditor({
+      const editor = new CoreEditor({
         theme: action.payload.theme,
         canvas: action.payload.canvas,
-        mode: state.editorLayoutMode
-          ? new modesMap[state.editorLayoutMode]()
-          : undefined,
         monomersLibraryUpdate: action.payload.monomersLibraryUpdate,
       });
+
+      state.editor = editor;
+      action.payload.onInit?.(editor);
```

Comment:
Looks like sort of side effect, isn't it supposed to be called within Editor upon it's instantiating?

Reply:
Will be considered as an improvement for both editors in master